src/__internal__/label/label.component.tsx

```diff
@@ -1,25 +1,73 @@
 import React, { useState, useContext } from "react";
-import PropTypes from "prop-types";
 import Help from "../../components/help";
 import StyledLabel, { StyledLabelContainer } from "./label.style";
 import ValidationIcon from "../validations/validation-icon.component";
 import IconWrapperStyle from "./icon-wrapper.style";
 import { InputContext, InputGroupContext } from "../input-behaviour";
+import { ValidationProps } from "../validations";
+import { IconType } from "../../components/icon";
 
-const shouldDisplayValidationIcon = ({ error, warning, info, disabled }) => {
+export interface LabelProps extends ValidationProps {
+  /** Label width */
+  width?: number;
+  /** Label alignment */
+  align?: "left" | "right";
+  /** When true, label is placed in line an input */
+  inline?: boolean;
+  /** Flag to configure component as mandatory */
+  isRequired?: boolean;
+  /** If true, the component will be disabled */
+  disabled?: boolean;
+  /** Flag to configure component as optional in Form */
+  optional?: boolean;
+  /** The unique id of the label element */
+  labelId?: string;
+  /** The unique id of the Help component tooltip, used for accessibility */
+  tooltipId?: string;
+  /** Children elements */
+  children?: React.ReactNode;
+  /** A message that the Help component will display */
+  help?: React.ReactNode;
+  /** Icon type */
+  helpIcon?: IconType;
+  /** Overrides the default tabindex of the Help component */
+  helpTabIndex?: string;
+  /** Whether to show the validation icon */
+  useValidationIcon?: boolean;
+  /** A string that represents the ID of another form element */
+  htmlFor?: string;
+  /** Padding right, integer multiplied by base spacing constant (8) */
+  pr?: 1 | 2;
+  /** Padding left, integer multiplied by base spacing constant (8) */
+  pl?: 1 | 2;
+  /** Id of the validation icon */
+  validationIconId?: string;
+}
+
+const shouldDisplayValidationIcon = ({
+  error,
+  warning,
+  info,
+  disabled,
+}: ValidationProps & { disabled?: boolean }) => {
   const validation = error || warning || info;
   return disabled ? false : typeof validation === "string";
 };
 
-const tooltipPosition = ({ error, warning, info, inline }) => {
-  return (error || warning || info) && inline === true ? "top" : "right";
+const tooltipPosition = ({
+  error,
+  warning,
+  info,
+  inline,
+}: ValidationProps & { inline?: boolean }) => {
+  return (error || warning || info) && inline ? "top" : "right";
```

Comment:
I think that the return type should be set as `TooltipPositions` type imported from `tooltip.config.ts`